src/postgres/commands/copyConnectionString.ts

```diff
@@ -17,13 +15,17 @@ export async function copyConnectionString(context: IActionContext, node: Postgr
         node = <PostgresDatabaseTreeItem>await ext.tree.showTreeItemPicker(PostgresDatabaseTreeItem.contextValue, context);
     }
 
-    const clientConfig: ClientConfig = await checkAuthentication(context, node);
+    await checkAuthentication(context, node);
 
-    const user: string = nonNullProp(clientConfig, 'user');
-    const password: string = nonNullProp(clientConfig, 'password');
-    const host: string = nonNullProp(clientConfig, 'host');
-    const port: string = String(nonNullProp(clientConfig, 'port'));
-    const connectionString = `postgres://${user}:${password}@${host}:${port}/'${node.databaseName}'`;
+    const parsedConnectionString = node.parent.connectionString;
+
+    if (!(parsedConnectionString.username && parsedConnectionString.password)) {
```

Comment:
I don't like the ordering of when/where you're calling `updateConnectionString`. You shouldn't need to call it here. There are two places you should call it:
1. Inside `PostgresServerTreeItem.getCredentials`. That way the connection string is always up to date with what you just retrieved from keytar and you won't have to call it in places like this.
1. Inside `enterPostgresCredentials.ts` - pretty self-explanatory

Also, I don't think you need an actual `updateConnectionString` method. Just make the two properties read-write and do something like this:
```
this.connectionString.username = username;
this.connectionString.password = password;
```